fdk.go

```diff
@@ -175,12 +176,16 @@ func doJSONOnce(handler Handler, ctx context.Context, in io.Reader, out io.Write
 	json.NewEncoder(out).Encode(jsonResponse)
 }
 
-func doHTTPOnce(handler Handler, ctx context.Context, in io.Reader, out io.Writer, buf *bytes.Buffer, hdr http.Header) {
-	// TODO we need to set deadline on ctx here (need FN_DEADLINE header)
-	// for now, just get a new ctx each go round
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+func CtxWithDeadline(parent context.Context, fnDeadline string) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(parent)
+	t, err := time.Parse(time.RFC3339, fnDeadline)
+	if err == nil {
+		ctx, cancel = context.WithDeadline(ctx, t)
+	}
+	return ctx, cancel
```

Comment:
slightly simpler control flow to do: 

```go
if err == nil {
  return context.WithDeadline(ctx, t)
}
return context.WithCancel(ctx)
```

it's also unconventional to name the context `parent` in the signature, please change (go lint would yell)